lib/alephant/broker/response/base.rb

```diff
@@ -48,10 +52,19 @@ def add_no_cache_headers
           log
         end
 
+        def add_etag_allow_header
+          headers.merge!("Access-Control-Expose-Headers" => "ETag")
+        end
+
         def component_not_modified(headers, request_env)
```

Comment:
By class level method I mean like `self.component_not_modified` - then you would call it in instance level methods via `self.class.component_not_modified`.

> Isn't the headers in this, driven by the instance?

No, as it's never called by an internal function - it's always called externally and the headers to be operated on are passed in.  Thats why it smells like a class method - it's operation is completely stateless.